### PR TITLE
ADEN-10917 Adding b_ias general IAS keyval

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: src/ad-services/ias-publisher-optimization/index.ts [hunk 1]
```diff
@@ -109,11 +122,23 @@ class IasPublisherOptimization {
...
 			brandSafetyKeys.forEach((key) => {
 				if (iasTargetingData.brandSafety[key]) {
 					context.set(`targeting.${key}`, iasTargetingData.brandSafety[key]);
+
+					if (
+						maxValue === '-1' ||
+						brandSafetyValuesLevel[maxValue] <
+							brandSafetyValuesLevel[iasTargetingData.brandSafety[key]]
+					) {
+						maxValue = iasTargetingData.brandSafety[key];
```
Comment (nandy-andy) [inline src/ad-services/ias-publisher-optimization/index.ts:136]: Can we unit test it?

Comment (dmnsobczak) [reply src/ad-services/ias-publisher-optimization/index.ts:136]: Yes! Unit tests added.

File: src/ad-services/ias-publisher-optimization/index.ts [hunk 2]
```diff
@@ -109,11 +122,23 @@ class IasPublisherOptimization {
...
 
 		if (iasTargetingData.brandSafety) {
+			let maxValue = '-1';
+
 			brandSafetyKeys.forEach((key) => {
 				if (iasTargetingData.brandSafety[key]) {
 					context.set(`targeting.${key}`, iasTargetingData.brandSafety[key]);
+
+					if (
+						maxValue === '-1' ||
```
Comment (Mariusz8Walczak) [inline src/ad-services/ias-publisher-optimization/index.ts:132]: it is a good practice to have max 3 nestings in one method

**Review by Mariusz8Walczak — COMMENTED** [review]
one thing for clean code, please fix it, or omit, you have a green light

File: src/ad-services/ias-publisher-optimization/index.ts (hunk 2 above)
Comment (dmnsobczak) [reply src/ad-services/ias-publisher-optimization/index.ts:132]: If I count it correctly it is 3rd nesting in current method, right?

Comment (Mariusz8Walczak) [reply src/ad-services/ias-publisher-optimization/index.ts:132]: Yep I can't count to 3, 2 is the limit

**Review by Mariusz8Walczak — APPROVED** [review]

**Pull request merged**